Functions section

jotas6 opened the pull request:
Content for the section about functions of the workshop

## Status
- [x] Edit README file
- [x] Code examples
- [x]  Instructor notes
- [x] Reference sheet
- [x] Schedule

jotas6 commented:
I just noticed that I didn't cover functions that modify some of the arguments (e.g `f!(x)`). Would it be a good idea to include this topic? I could fit it in `02-advanced.jl` at the beginning or create a new script for it.

storopoli commented on 03-functions/01-syntax.jl:
```diff
@@ -0,0 +1,63 @@
+add(2, 3)
+add(1, 10)
+
+f(x) = x^2 # Natural way to define math functions
```
this should be named compact function assignment

storopoli commented on 03-functions/01-syntax.jl:
```diff
@@ -0,0 +1,63 @@
+
+## Return multiple values
+function even_greater_than_10(number)
+
```
```suggestion
```

storopoli commented on 03-functions/01-syntax.jl:
```diff
@@ -0,0 +1,63 @@
+    greater = number > 10
+
+    return even, greater # Return multiple values with return1, return2, return3, ...
+
```
```suggestion
```

storopoli commented on 03-functions/01-syntax.jl:
```diff
@@ -0,0 +1,63 @@
+even = checks[1]
+greater = checks[2]
+
+## Tip: compact way to get multiple values
```
```suggestion
## Tip: useful way to unpack multiple values
```

storopoli commented on docs/index.md:
```diff
@@ -12,7 +12,7 @@ This workshop is an **introduction to the Julia programming language,** covering
     - Strings
     - Containers
-
+    
```
```suggestion

```

storopoli commented on docs/instructors.md:
```diff
@@ -96,6 +96,43 @@ First, show that macros start with `@` and quickly go over the examples. Next, s
+
+Start by going over `01-syntax.jl` as this script will cover the basics of defining and calling functions. First, show the standard way
+of creating functions with the `function` keyword, and then show how a function can be called once it has been defined. Use the examples
+to show that a function can have a single argument, multiple arguments, or no arguments at all. Next, go over the single line syntax to
```
Say about the compact assignment form

storopoli commented on docs/reference.md:
```diff
@@ -63,6 +63,33 @@ be replaced with underscores (`_`).
+### Part 3: functions
+
+- Complex functions can be created with the `function` keyword, while simpler ones can be 
+more conveniently defined using the single line syntax (e.g `f(x) = x^2`).
```
compact assignment form

storopoli reviewed (commented):
some comments

jotas6 commented:
@storopoli I think I have covered the comments. I changed "single line syntax" to "compact assignment form" in all the places I could find it, and removed some whitespace from a function definition.
